public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistory.tsx

```diff
@@ -0,0 +1,383 @@
+import { css } from '@emotion/css';
+import React, { useCallback, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useMeasure } from 'react-use';
+
+import { GrafanaTheme2, TimeRange } from '@grafana/data';
+import { isFetchError } from '@grafana/runtime';
+import { SceneComponentProps, SceneObjectBase, sceneGraph } from '@grafana/scenes';
+import {
+  Alert,
+  Button,
+  Field,
+  Icon,
+  Input,
+  Label,
+  LoadingBar,
+  Stack,
+  Text,
+  Tooltip,
+  useStyles2,
+  withErrorBoundary,
+} from '@grafana/ui';
+import { EntityNotFound } from 'app/core/components/PageNotFound/EntityNotFound';
+import { Trans, t } from 'app/core/internationalization';
+import {
+  GrafanaAlertStateWithReason,
+  isAlertStateWithReason,
+  isGrafanaAlertState,
+  mapStateWithReasonToBaseState,
+  mapStateWithReasonToReason,
+} from 'app/types/unified-alerting-dto';
+
+import { stateHistoryApi } from '../../../api/stateHistoryApi';
+import { GRAFANA_RULES_SOURCE_NAME } from '../../../utils/datasource';
+import { stringifyErrorLike } from '../../../utils/misc';
+import { hashLabelsOrAnnotations } from '../../../utils/rule-id';
+import { AlertLabels } from '../../AlertLabels';
+import { CollapseToggle } from '../../CollapseToggle';
+import { LogRecord } from '../state-history/common';
+import { useRuleHistoryRecords } from '../state-history/useRuleHistoryRecords';
+
+const LIMIT_EVENTS = 250;
+
+const HistoryEventsList = ({ timeRange }: { timeRange?: TimeRange }) => {
+  const styles = useStyles2(getStyles);
+
+  // Filter state
+  const [eventsFilter, setEventsFilter] = useState('');
+  // form for filter fields
+  const { register, handleSubmit, reset } = useForm({ defaultValues: { query: '' } }); //  form for search field
+  const from = timeRange?.from.unix();
+  const to = timeRange?.to.unix();
+  const onFilterCleared = useCallback(() => {
+    setEventsFilter('');
+    reset();
+  }, [setEventsFilter, reset]);
+
+  const {
+    data: stateHistory,
+    isLoading,
+    isError,
+    error,
+  } = stateHistoryApi.endpoints.getRuleHistory.useQuery(
+    {
+      from: from,
+      to: to,
+      limit: LIMIT_EVENTS,
+    },
+    {
+      refetchOnFocus: true,
```

Comment:
nit: These would also not be really useful as the data will never change even if the window is re-focussed or the network is re-connected – the stale data should still be valid :)

Reply:
💯